.github/workflows/build.yml

```
@@ -6,6 +6,155 @@
+      - uses: gitleaks/gitleaks-action@v2
+        with:
+          args: detect --redact --report-path=gitleaks-report.json --exit-code=1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITLEAKS_LICENSE: ${{ secrets.GITLEAKS_LICENSE }}
+
+      - name: Upload SARIF
+        if: always() && hashFiles('gitleaks-report.json') != ''
+        uses: github/codeql-action/upload-sarif@v3
+        with:
+              sarif_file: gitleaks-report.json
```

YAMLlint (1.37.1)

[warning] wrong indentation: expected 10 but found 14

(indentation)